### DISCUSSION TITLE
Registration captcha issue

### DISCUSSION THREAD

**Discussion by the01** [AsamK/signal-cli#628] [Q&A] [answered]
Hi,
I'm having trouble registering my landline.
I have a docker container based on Debian with signal-cli downloaded from the git releases.

I got the captcha from the website just fine (`signalcaptcha://03AGdBq ... QYf8SBvg`)
but when I run the register command
```
signal-cli --config signal -u +xxxxxxx register --voice  --captcha 03AGdBq ... QYf8SBvg
```
I always get `Request verify error: Bad response: 400` and after some attempts `Request verify error: Rate limit exceeded: 413`.

I even tried with different signal-cli versions down to 0.7.4, but same results.

Any ideas what the issue might be?

**Comment by AsamK** [comment, marked as answer]
Not sure what's going on there. Does running the verify command with `--verbose` output more information?

**Reply by the01** [reply]
No, just the `Request verify error` line

**Reply by the01** [reply]
I just tried it on an ubuntu 20.04 (wsl) with the same result.
I'm assuming this message is still in regards to the captcha and not the phone line itself?

**Reply by AsamK** [reply]
It shouldn't be an issue with the captcha, that would return a 402 response.
I've had a look at the Signal-Server code and a 400 response is only returned if the number is invalid or the language code can't be parsed.
- Can you check if your number is recognized as valid by libphonenumber ([Test site](https://libphonenumber.appspot.com/))
- What's value of the `LANG` and `LANGUAGE` environment variables on your system?

**Reply by the01** [reply]
wow, thank you so much!!
`LANG` was messed up and setting the correct value finally did the trick 😄 😄 🎉 

Out of curiosity, did I miss this somewhere in the docs?

**Reply by AsamK** [reply]
No, it's not documented, signal-cli internally includes the system language in the register command.
I guess I'll add a check or workaround for that. Also curious, what was the value of `LANG`?

**Reply by the01** [reply]
Well, glad that I at least read the documentation carefully then 😄 

Before it was a generic `C.UTF-8` from my base docker template and the country specific `de_AT.UTF-8` was the correct one

**Reply by fabbra** [reply]
I have a very similar issue to register a Swiss landline number using `signal-cli 0.10.5` saying `NonNormalizedPhoneNumberException: StatusCode: 400`, see [here](https://github.com/AsamK/signal-cli/issues/958#issue-1244070642).
